spec.md

```diff
@@ -3,193 +3,138 @@ title: "HTTP API V2"
+
+### Table of Contents
+
+- [Introduction](spec.md)
+- [Notational Conventions](#notational-conventions)
+- [Historical Context](#historical-context)
+- [Scope](#scope)
+   - [Future](#future)
+- [Use Cases](#use-cases)
+   - [Image Verification](#image-verification)
+   - [Resumable Push](#resumable-push)
+   - [Resumable Pull](#resumable-pull)
+   - [Layer Upload De-duplication](#layer-upload-de-duplication)
+- [Changes](#changes)
+- [Overview](#overview)
+   - [Errors](#errors)
+   - [API Version Check](#api-version-check)
+   - [Content Digests](#content-digests)
+   - [Pulling An Image](#pulling-an-image)
+   - [Pushing An Image](#pushing-an-image)
+   - [Listing Repositories](#listing-repositories)
+   - [Listing Image Tags](#listing-image-tags)
+   - [Deleting an Image](#deleting-an-image)
+- [Detail](#detail)
+   - [Errors](#errors-2)
+   - [Base](#base)
+   - [Tags](#tags)
+   - [Manifest](#manifest)
+      - [GET Manifest](#get-manifest)
+      - [PUT Manifest](#put-manifest)
+      - [DELETE Manifest](#delete-manifest)
+   - [Blob](#blob)
+      - [GET Blob](#get-blob)
+         - [Fetch Blob](#fetch-blob)
+         - [Fetch Blob Part](#fetch-blob-part)
+      - [DELETE Blob](#delete-blob)
+   - [Initiate Blob Upload](#initiate-blob-upload)
+      - [POST Initiate Blob Upload](#post-initiate-blob-upload)
+         - [Initiate Monolithic Blob Upload](#initiate-monolithic-blob-upload)
+         - [Initiate Resumable Blob Upload](#initiate-resumable-blob-upload)
+         - [Mount Blob](#mount-blob)
+   - [Blob Upload](#blob-upload)
+      - [GET Blob Upload](#get-blob-upload)
+      - [PATCH Blob Upload](#patch-blob-upload)
+      - [PUT Blob Upload](#put-blob-upload)
+      - [DELETE Blob Upload](#delete-blob-upload)
+   - [Catalog](#catalog)
+      - [GET Catalog](#get-catalog)
+
+## Notational Conventions
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" are to be interpreted as described in [RFC 2119](http://tools.ietf.org/html/rfc2119) (Bradner, S., "Key words for use in RFCs to Indicate Requirement Levels", BCP 14, RFC 2119, March 1997).
+
+The key words "unspecified", "undefined", and "implementation-defined" are to be interpreted as described in the [rationale for the C99 standard][c99-unspecified].
+
+An implementation is not compliant if it fails to satisfy one or more of the MUST, MUST NOT, REQUIRED, SHALL, or SHALL NOT requirements for the protocols it implements.
+An implementation is compliant if it satisfies all the MUST, MUST NOT, REQUIRED, SHALL, and SHALL NOT requirements for the protocols it implements.
+
+## Historical Context
+
+For relevant details and a history leading up to this specification, please see the following issues:
```

Comment:
oh, maybe historical defacto info could just be more added here.

Reply:
`Background`? This needs to be filled out

Reply:
Adding a TODO